src/rcc.rs

```diff
@@ -121,12 +123,16 @@ pub struct AHB {
 }
 
 impl AHB {
-    pub(crate) fn enr(&mut self) -> &rcc::AHBENR {
+    pub(crate) fn new() -> Self {
+        Self { _0: () }
+    }
+
+    pub(crate) fn enr(&self) -> &rcc::AHBENR {
         // NOTE(unsafe) this proxy grants exclusive access to this register
         unsafe { &(*RCC::ptr()).ahbenr }
     }
 
-    pub(crate) fn rstr(&mut self) -> &rcc::AHBRSTR {
```

Comment:
Even though taking a shared reference is ok, I'd like to keep and exclusive &mut reference, as it is allowed to change the register through the reference (despite being a shared immutable reference).